### DISCUSSION TITLE
State timeline needs vertical sizing control (min/max height? scrollbar?)

### DISCUSSION THREAD

**Discussion by james00002** [grafana/grafana#40817] [Dashboards, Panels & Visualizations]
The new State Timeline panel is really cool, but it has a glaring weakness.

There is no way to control the vertical sizing of the state bars, it always tries to fit all bars into the panel's vertical limits with no respect to how many of them there are.

That means for a dashboard that may dynamically show few or many bars depending on variable filters, you might end up with 2 or 3 gigantic super-thick bars, or hundreds of super thin bars with the text overlapping and unreadable.

A good solution would be to add an option to choose a minimum and maximum bar height. Both could be optional, defaulting to the current auto-sizing behavior if unspecified.

Enabling a minimum bar size would also mean a scrollbar would have to appear if there are too many series to fit vertically, similar to how the Table panel has a scrollbar

I don't know if it would be better for min/max heights to be specified in px, or in % of the panel height 🤷  but either one would still be very useful

**Comment by leeoniya** [comment]
duplicate of https://github.com/grafana/grafana/discussions/38303

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
